.github/workflows/publish.yml

```diff
@@ -0,0 +1,70 @@
+name: publish
+
+on:
+  pull_request:
+    branches: [master]
+  push:
+    branches: [master]
+
+# Sets permissions of the GITHUB_TOKEN to allow deployment to GitHub Pages
+permissions:
+  contents: read
+  pages: write
+  id-token: write
+
+# Allow one concurrent deployment
+concurrency:
+  group: 'pages'
+  cancel-in-progress: true
+
+jobs:
+  checks:
+    if: github.event_name != 'push'
+    runs-on: ubuntu-latest
+    steps:
+      - uses: actions/checkout@v2
+        with:
+          submodules: recursive
+      - name: Setup mdBook
+        uses: peaceiris/actions-mdbook@v1
+        with:
+          mdbook-version: 'latest'
+      - name: Setup wasm-pack
+        uses: jetli/wasm-pack-action@v0.4.0
+        with:
+          version: 'latest'
+      - name: Install Task
+        uses: arduino/setup-task@v1
+      - name: Run Build
+        run: task
+  gh-release:
+    if: github.event_name != 'pull_request'
+    runs-on: ubuntu-latest
+    environment:
+      name: github-pages
```

Comment:
Do we need to create this environment first? One thing I'd like is that it's not necessarily deployed on every commit (with environments, one can select the reviewers who will approve or deny a particular deployment)

Reply:
I'm a bit confused as I'm not too familiar with the extra features of GitHub environments (and have only been using them as shells for deploying to GitHub pages) - can't you establish extra publishing features in the repository settings? ([GitHub Documentation link](https://docs.github.com/en/actions/deployment/targeting-different-environments/using-environments-for-deployment)).

Reply:
Correct, I'm just confirming that before this is merged, we should add the environment in the repository settings, right?

Reply:
I don't think the environment needs to be configured in repository settings, but adding your respective `CNAME` file and changing github pages to use Github Actions are the required changes.
<img width="827" alt="image" src="https://user-images.githubusercontent.com/26509014/216081609-9be9bf85-4c6b-4f85-bbf6-614dd273322b.png">